src/fields.tsx

```diff
@@ -33,5 +33,5 @@ export function numberField(
 }
 
 export function arrayField<T>(props?: FieldProps<T[]>): FieldDefinition<T[]> {
-  return field({ default: [], ...props })
+  return field({ default: [], rules: [required, nonEmpty], ...props })
```

Comment:
Hmm I don't think we should assume that all use cases require a non-empty array. For example, it'd be fine for a multi-select style field to accept an empty array as valid input.

Instead we could add a `nonEmptyArrayField` convenience method, so we have both use-cases covered.

Reply:
Makes sense 👍